### PR TITLE
Added API for packages, repos and resource usage statistics.

### PR DESCRIPTION
**Pull request by cmaglie**

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -109,91 +105,89 @@ type UploadPayload struct {
...
-		}
-
-		// Stop and delete if existing
-		if sketch, ok := status.Sketches[info.ID]; ok {
-			err = applyAction(sketch, "STOP", status)
-			if err != nil {
-				status.Error("/upload", errors.Wrapf(err, "stop pid %d", sketch.PID))
-				return
-			}
+func (status *Status) UploadEvent(client mqtt.Client, msg mqtt.Message) {
```
[the remark this answers is not in the document]
Comment (cmaglie) [reply handlers.go]: I've just simplified the usage of callback by removing one indirection, previously `UploadCB` returned an `mqtt.MessageHandler` to be registered with mqtt lib

```go
func UploadCB(status *Status) mqtt.MessageHandler {
 	return func(client mqtt.Client, msg mqtt.Message) {
               [...do things with client and msg...]
```

now `UploadCB` **is** the `mqtt.MessageHandler`:

```go
func (status *Status) UploadEvent(client mqtt.Client, msg mqtt.Message) {
       [...do things with client and msg...]
```

We are dependent on `mqtt` exactly as before...

If we want to be protocol-agnostic we should actually try to implement another protocol to see where the hypotetical interface should be placed.

[the remark this answers is not in the document]
Comment (cmaglie) [reply handlers.go]: Too see the real extension of this commit is better to filter whitespace-diffs:
https://github.com/arduino/arduino-connector/pull/28/commits/d324c03b19c9200ba16d083d458c58cbad842f93?w=1

**Comment by cmaglie** [comment]
> should we ignore case for search terms? @mastrolinux

packages are all lower-case (I guess it is by policy on debian?) unless the user types upper case chars in the search box there should be no problems.

**Comment by mastrolinux** [comment]
the frontend directly converts uppercase to lowercase, that was done by @smellai I am going to merge as it is now

**Pull request merged**